DigitalLearningSolutions.Data/DataServices/CourseDataService.cs

```diff
@@ -659,6 +684,28 @@ public void UpdateCourseOptions(CourseOptions courseOptions, int customisationId
             );
         }
 
+        public Dictionary<int, int> GetNumsOfRecentProgressRecordsForBrand(int brandId, DateTime threeMonthsAgo)
+        {
+            var query = connection.Query(
+                @"SELECT Applications.ApplicationID, COUNT(Progress.ProgressID) AS Num_Recent_Progress_Records
+                    FROM Applications
+                        INNER JOIN Customisations ON Applications.ApplicationID = Customisations.ApplicationID
+                        INNER JOIN Progress ON Customisations.CustomisationID = Progress.CustomisationID
+                    WHERE Applications.BrandID = @brandId
+                      AND Applications.Debug = 0
+                      AND Applications.ArchivedDate IS NULL
+                      AND Progress.SubmittedTime > @threeMonthsAgo
+                    GROUP BY Applications.ApplicationID
+                    ORDER BY Num_Recent_Progress_Records DESC",

```

Comment:
Can we move the ordering into the service? I don't think we order queries anywhere else, so it would be helpful to keep a consistent approach of "This will be ordered by id or not at all" in data service queries, so our services know what they can and can't expect.